src/Lecture2.hs

```diff
@@ -74,7 +84,10 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: (Ord a) => Int -> [a] -> (Maybe a, [a])
+removeAt index list
+           | index < 0 || index >= length list = (Nothing, list)
```

Comment:
Function `length` hangs on infinite lists. And because of that, your implementation of `removeAt` doesn't work on infinite lists ♾️